common/src/main/java/org/apache/apex/common/util/SystemPropertyReader.java

```diff
@@ -0,0 +1,48 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.apex.common.util;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SystemPropertyReader
+{
+  public static long getLong(String propertyName, long defaultValue, long minValue, long maxValue)
+  {
+    String property = System.getProperty(propertyName);
+    long result = defaultValue;
+    if (property != null) {
+      try {
+        long value = Long.decode(property);
+        if (value <= minValue || value >= maxValue) {
```

Comment:
This has introduced a regression. Caller calls with minValue of 0 which is disallowed because of `value <= minValue`. 

In any case, since we are creating a Util class/method we should have Javadocs to describe the method and params and in this case whether the min/max values are inclusive/exclusive.